src/commonMain/kotlin/exchange.dydx.abacus/processor/wallet/account/SubaccountProcessor.kt

```diff
@@ -49,24 +58,54 @@ internal open class SubaccountProcessor(parser: ParserProtocol) : BaseProcessor(
         ),
     )
 
-    internal fun subscribed(
+    internal fun processSubscribed(
+        existing: InternalSubaccountState,
+        content: Map<String, Any>,
+        height: BlockAndTime?,
+    ): InternalSubaccountState {
+        return processSocket(existing, content, true, height)
+    }
+
+    internal fun subscribedDeprecated(
         existing: Map<String, Any>?,
         content: Map<String, Any>,
         height: BlockAndTime?,
     ): Map<String, Any> {
-        return socket(existing, content, true, height)
+        return socketDeprecated(existing, content, true, height)
+    }
+
+    internal fun processChannelData(
+        existing: InternalSubaccountState,
+        content: Map<String, Any>,
+        height: BlockAndTime?,
+    ): InternalSubaccountState {
+        return processSocket(existing, content, false, height)
     }
 
     @Suppress("FunctionName")
-    internal fun channel_data(
+    internal fun channel_dataDeprecated(
         existing: Map<String, Any>?,
         content: Map<String, Any>,
         height: BlockAndTime?,
     ): Map<String, Any> {
-        return socket(existing, content, false, height)
+        return socketDeprecated(existing, content, false, height)
     }
 
-    internal open fun socket(
+    internal open fun processSocket(
+        existing: InternalSubaccountState,
+        content: Map<String, Any>,
+        subscribed: Boolean,
+        height: BlockAndTime?,
+    ): InternalSubaccountState {
+        val fillResponse = Json.decodeFromString<IndexerFillResponse?>(content.toJson())
```

Comment:
need try catch here

Reply:
<img width="738" alt="Screenshot 2024-07-09 at 6 47 37 PM" src="https://github.com/dydxprotocol/v4-abacus/assets/163016611/4d44ac73-8ab3-4f18-9c1c-8113f5067644">

Reply:
My follow-up PR already has this handled.